src/main/java/com/stumeet/server/file/adapter/out/S3ImageStorageAdapter.java

```diff
@@ -48,21 +51,49 @@ public FileUrl uploadImageFile(MultipartFile file, String directoryPath) {
 		try {
 			s3Client.putObject(
 				objectRequest,
-				RequestBody
-					.fromInputStream(file.getInputStream(), file.getSize()));
+				RequestBody.fromInputStream(file.getInputStream(), file.getSize()));
+
+			return getS3FileUrl(key);
 		} catch (IOException e) {
+			throw new BusinessException(e, ErrorCode.FILE_IO_ERROR);
+		} catch (Exception e) {
 			throw new BusinessException(e, ErrorCode.UPLOAD_FILE_FAIL_ERROR);
 		}
+	}
 
-		return getS3FileUrl(key);
+	private FileUrl getS3FileUrl(String key) {
+		return new FileUrl(String.format("%s/%s", endpoint, key));
 	}
 
 	@Override
 	public List<FileUrl> uploadImageFiles(List<MultipartFile> multipartFileList, String directoryPath) {
 		throw new NotImplementedException();
 	}
 
-	private FileUrl getS3FileUrl(String key) {
-		return new FileUrl(String.format("%s/%s", endpoint, key));
+	@Override
+	public void deleteFolder(String folder) {
+		List<ObjectIdentifier> objectIdentifiers = getObjectIdentifiers(folder);
+
+		if (!objectIdentifiers.isEmpty()) {
+			DeleteObjectsRequest deleteObjectsRequest = DeleteObjectsRequest.builder()
+				.bucket(bucket)
+				.delete(builder -> builder.objects(objectIdentifiers))
+				.build();
+
+			s3Client.deleteObjects(deleteObjectsRequest);
+		}
+	}
```

Comment:
P5 : 취향의 차이가 있는 부분이라고 생각하지만 저는 개인적으로 특정 조건에 대해 더 이상 로직을 실행시키지 않는 경우  Early return을 사용하는 방식을 선호하는데 찬미님 생각이 궁금합니다 🤔  
```suggestion
	@Override
public void deleteFolder(String folder) {
    List<ObjectIdentifier> objectIdentifiers = getObjectIdentifiers(folder);

    // 식별되는 오브젝트가 없는 경우 바로 리턴
    if (objectIdentifiers.isEmpty()) {
        return;
    }

    DeleteObjectsRequest deleteObjectsRequest = DeleteObjectsRequest.builder()
        .bucket(bucket)
        .delete(builder -> builder.objects(objectIdentifiers))
        .build();

    s3Client.deleteObjects(deleteObjectsRequest);
}
```

Reply:
저도 return 문과 고민하다가 조건문을 사용했는데, 갑자기 return 구문을 사용하는 경우 코드 흐름이 끊기고 권한을 남용할 수 있겠다라는 생각이 들었습니다!

하지만 여러 논의를 찾아보고 제안해주신 예제를 보니 조건에 부합할 경우 조기에 종료시키는게 가독성 면에서 좋아보이는 것 같습니다!